### ISSUE TITLE
Support multiple eBPF programs in the same section

### ISSUE THREAD

**Issue by dthaler** [microsoft/ebpf-for-windows#3542]
### Describe the feature you'd like supported

Quentin Monnet pointed out two years ago that libbpf v1.0 now allows multiple programs in the same section ([Andrii's blog again](https://nakryiko.com/posts/libbpf-v1/#bpf-program-sec-annotation-streamlining))).

Also, the prog_array.o sample in the ebpf-samples repository does this.

### Proposed solution

Using a section name is not sufficient.  Instead use the C function name.

Support is blocked on https://github.com/vbpf/ebpf-verifier/pull/638 being merged first.

**Comment by dthaler** [comment]
Duplicate of #1440
